### PR TITLE
feat(web): empty and loading states for local studio

### PR DESCRIPTION
**Pull request by ainouzgali**
### What changed? Why was the change needed?

- Loading state for Workflows detail page / landing page
- Display for having no workflows
- Display for disconnected state
- Refactor to have connection status hook

### Open Questions

- [ ] How do we want to handle when the user gets disconnected? It seems that auto-redirecting to that page would be frustrating UX
- [ ] Do we really need the skeleton loading state in the sidebar when disconnected?

### Screenshots

Loading 
Screenshot 2024-06-27 at 6 31 38 PM

Empty / No workflows
Screenshot 2024-06-27 at 6 32 21 PM

Disconnected
Screenshot 2024-06-27 at 6 32 47 PM

### CODE REVIEW

File: apps/web/src/studio/components/workflows/landing-page/WorkflowPanelEmptyState.tsx [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+      <br />
+      Get started by adding your first workflow in your local environment.
+    </WorkflowPlaceholderPageContent>
+  );
+}
+
+function WorkflowPanelUnconnectedContent() {
+  return (
+    <WorkflowPlaceholderPageContent docsButtonLabel="See our troubleshooting guide">
+      Local environment disconnected from endpoint URL.
```
Comment (SokratisVidros) [inline apps/web/src/studio/components/workflows/landing-page/WorkflowPanelEmptyState.tsx]:
```suggestion
      Local environment disconnected from Novu Bridge URL.
```

File: apps/web/src/studio/components/workflows/landing-page/WorkflowPanelEmptyState.tsx [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+  return (
+    <WorkflowPlaceholderPageContent docsButtonLabel="Learn more in the docs">
+      A workflow holds the entire flow of steps that are sent to the subscriber.
+      <br />
+      Get started by adding your first workflow in your local environment.
+    </WorkflowPlaceholderPageContent>
+  );
+}
+
+function WorkflowPanelUnconnectedContent() {
```
Comment (SokratisVidros) [inline apps/web/src/studio/components/workflows/landing-page/WorkflowPanelEmptyState.tsx]:
```suggestion
function WorkflowPanelDisconnectedContent() {
```

File: apps/web/src/studio/hooks/useBridgeConnectionStatus.ts [hunk 3]
```diff
@@ -0,0 +1,24 @@
+import { useMemo } from 'react';
+import { ConnectionStatus } from '../types';
+import { useHealthCheck } from './useBridgeAPI';
+
+export const useBridgeConnectionStatus = () => {
+  const { data, isFetching, error, bridgeURL } = useHealthCheck();
+
+  const status = useMemo<ConnectionStatus>(() => {
+    if (isFetching) {
+      return 'loading';
```
Comment (SokratisVidros) [inline apps/web/src/studio/hooks/useBridgeConnectionStatus.ts:10]: Please use the connection status enum.

**Review by SokratisVidros — APPROVED** [review]

File: apps/web/src/studio/hooks/useBridgeConnectionStatus.ts (hunk 3 above)
Comment (antonjoel82) [reply apps/web/src/studio/hooks/useBridgeConnectionStatus.ts:10]: @SokratisVidros Where is this enum? This is taken from the hook you wrote

Comment (antonjoel82) [reply apps/web/src/studio/hooks/useBridgeConnectionStatus.ts:10]: @SokratisVidros That is being used, but it's not an enum

Comment (antonjoel82) [reply apps/web/src/studio/hooks/useBridgeConnectionStatus.ts:10]: @SokratisVidros  That is being used, but it's not an enum!

Comment (antonjoel82) [reply apps/web/src/studio/hooks/useBridgeConnectionStatus.ts:10]: It's also typed based on the useMemo

**Pull request merged**